### PR TITLE
getBalances fixes:

### PR DESCRIPTION
**Pull request by darkdarkdragon**
obey limit, do not return XRP if currency or issuer specified

### CODE REVIEW

**Comment by clark800** [comment]
Can you add a test case to cover this?

**Comment by darkdarkdragon** [comment]
@clark800 done

File: test/api-test.js [hunk 1]
```diff
@@ -196,9 +196,45 @@ describe('RippleAPI', function() {
...
+    it('getBalances', function() {
+      return this.api.getBalances(address).then(
+        _.partial(checkResult, responses.getBalances, 'getBalances'));
+    });
+
+    it('getBalances - limit', function() {
+      const options = {
+        limit: 3
+      };
+      const response = responses.getBalances.slice(0, 3);
```
Comment (clark800) [inline test/api-test.js]: Can you rename to `expected` or `expectedResponse`?

**Comment by darkdarkdragon** [comment]
@clark800 done

**Comment by clark800** [comment]
LGTM

**Pull request merged**